### ISSUE TITLE
Add support for setting path in GM_download

### ISSUE THREAD

**Issue by RezqZiyad** [violentmonkey/violentmonkey#1349]
Hi, Why don't you add the ability to set optional "managing downloads" permission and the functionality to support setting path in GM_download like it is in TM?
![image](https://user-images.githubusercontent.com/29045522/130663926-e93cc589-9c49-4d22-b41e-423e6d885ab1.png)

**Comment by tophf** [comment]
Yeah, setting a path is a nice thing. What happens in TM if the mode is not `Browser API` and GM_download is used with a path?

**Comment by RezqZiyad** [comment]
It just ignores the path. The file is saved to the default download folder.

**Comment by tophf** [comment]
I thought maybe we should surface an error somehow to let the user or the script know that VM has an option now. Well, if TM conceals it then we'll do the same.

**Comment by RezqZiyad** [comment]
TM really doesn't support paths by default. You have to choose Browser API (appears in the screenshot) and then asks to update permissions. I also updated my previous comment.

**Comment by tophf** [comment]
Yeah I know that it asks for a permission, I just wondered about improving it, but I guess there's no need.

**Comment by RezqZiyad** [comment]
Got it! I think if an author wanted to use paths they will notify the user. A user wouldn't notice errors. May be you should add a note on documentation for authors and tip next to the option for them all like TM does: 
![image](https://user-images.githubusercontent.com/29045522/130691217-6a28ed7c-3b84-4194-b71d-8260766c4ff7.png)

**Comment by jesus2099** [comment]
Thanks @tophf for adding `saveAs` support request! :)

**Comment by qinxs** [comment]
Are there any plans to implement this feature in the near future?

**Comment by cunlem** [comment]
Is this a standard API or will it potentially only work in Chrome-based browsers?

**Comment by mikelei8291** [comment]
@cunlem It works in both Chrome and Firefox: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/downloads/download#browser_compatibility

**Comment by redfieldw** [comment]
I really hope that the Violentmonkey can support paths in GM_download

**Comment by tostr7191** [comment]
Is there a specific reason why the DOM download is used as opposed to `browser.downloads.download`? The latter would help alleviate some of the issues, like overwriting behaviour, showing saveAs, and also subdirectories.

I would be very interested in understanding if this is a feature that will be implemented, and what the reasoning is, in case it won't be implemented. No offense or criticism intended whatsoever, just curious since it seems actually simpler to implement than the current implementation. The only advantages (from my limited understanding) of the current implementation seem to be that there is no need to request another permission and that you can grab data from cache should it reside there (at least I think that is how it works).

However your reasoning may be, I would also add a vote for it, since it would help me greatly to sort my downloads into a more reasonable structure without manual intervention.

**Comment by tophf** [comment]
Well the implementation is not exactly trivial and as I don't need this feature I have zero motivation, same for @gera2ld I guess. This is an open source project so maybe someone else will open a PR with their implementation. Eventually if I'm still around it's likely that I run into needing it myself and I'll implement it then.
